exchange/bidder.go

```diff
@@ -501,7 +507,20 @@ func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.Reques
 }
 
 func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.RequestData, logger util.LogMsg) *httpCallInfo {
-	httpReq, err := http.NewRequest(req.Method, req.Uri, bytes.NewBuffer(req.Body))
+	var requestBody []byte
+
+	switch bidder.config.EndpointCompression {
+	case "GZIP":
+		var b bytes.Buffer
```

Comment:
In config "GZIP" can be also specified as "gzip" or even "gZip". This case will never work.
Please replace switch statement to switch strings.ToUpper(bidder.config.EndpointCompression) {.
Also I recommend to extract "GZIP" to constant.